app-web/src/components/Home/ResourcePreview.js

```diff
@@ -84,49 +88,92 @@ export const TEST_IDS = {
+  const [activeFilters, setActiveFilters] = useState([]);
 
+  // this effect extracts initial state from the url parameter
+  // we only run this when location changes from doing searches
   useEffect(() => {
-    setResources(resources);
+    const windowHasFilters = Object.prototype.hasOwnProperty.call(queryParam, FILTER_QUERY_PARAM);
+    const filterQueryParams = windowHasFilters
+      ? decodeURIComponent(queryParam[FILTER_QUERY_PARAM]).split(',')
+      : [];
+    const validFilters = intersection(filtersList, filterQueryParams);
+    if (validFilters.length !== activeFilters.length) {
+      // make sure filters are treated as an array
+      // if only one filter is passed as param ?f= it will be a string
+      setActiveFilters(validFilters);
+    } else if (activeFilters.length !== 0) {
+      setActiveFilters([]);
+    }
+    // eslint-disable-next-line
+  }, [location.search]);
 
-    return () => {
-      setResources(null);
-    };
-  }, [resources]);
+  const isAllToggled = activeFilters.length === 0 || activeFilters.length === filters.length;
+
+  const filteredResources = isAllToggled
+    ? resources
+    : resources.filter(resource => activeFilters.includes(resource.fields.resourceType));
 
   //sets the amount of resources to show, allowing users to 'see more' if its appropriate
   const updateCount = () => {
     //show 6 more results
     setSeeMore(true);
     setCount(showCount + extraItemsToShow);
-    if (resourcesToShow.length <= showCount) {
+    if (resources.length <= showCount) {
       setSeeMore(false); //hide the 'see more results' when there isn‘t more to show
     }
   };
 
   // This filters what results we are showing based on the given filter coming from user interaction with the ResourcePills
-  const resourceFilter = filterName => {
-    //filter the results based on given filter, update the resources and active filter
-    let filteredResources = [];
-    let newPillList = togglePills(filterName, activeFilters);
-    setActiveFilters(newPillList);
-
-    filteredResources = resources.filter(resource =>
-      newPillList.includes(resource.fields.resourceType),
-    );
-    if (newPillList.includes('All')) {
-      setResources(resources);
-    } else {
-      setResources(filteredResources);
-    }
-    //reset the amount of resources to show
+
+  const resourceFilterClicked = filterName => {
+    const { pathname } = location;
+    const toggledFilters = handleFilterToggle(filterName, activeFilters);
+
+    const searchMap = {
+      ...queryParam,
+      [FILTER_QUERY_PARAM]: toggledFilters.length === filtersList.length ? [] : toggledFilters,
+    };
+    const searchString = queryString.stringify(searchMap);
+
     setCount(amountToShow);
+    // replacing state so that back button doesn't get poluted with filter changes
+    // also this will trigger the url to change but will not cause a page refresh unlike gatsby.navigate
+    window.history.replaceState({}, 'foo', `${pathname}?${searchString}`);
```

Comment:
Used window.replacestate instead of gatsby navigate due to auto scolling in the naviagte function